### PR TITLE
Rename oecert tool

### LINKED ISSUE #4012: Rename oecert tool
Since `oecert` is used to generate not only certificates but also reports and evidence in different formats, and we have another tool `oeverify` for verifying evidence, we would like to propose a new name for it, e.g., `oegen`. See the discussion in [PR 4008](https://github.com/openenclave/openenclave/pull/4008#issuecomment-839927526).

### PR DESCRIPTION
**Pull request by ryanhsu19**
1. Propose `oegenerate` as the new name for the `oecert` tool (`oegen` was the original proposal but we think that "generate" sounds more complete than "gen"), while the other similar tool, `oecertdump`, is still `oecertdump`.
2. Fix some typos (rename `sgx_asem_env` to `sgx_aesm_env`) in `tools/oegenerate/host/host.cpp`.
3. Fix #4012.

Signed-off-by: Ryan Hsu <ryhsu@microsoft.com>

### CODE REVIEW

**Review by mingweishih — APPROVED** [review]

**Review by mingweishih — COMMENTED** [review]
Please update the change log as well.

**Review by qiucwang — APPROVED** [review]
LGTM. Thanks!

File: CHANGELOG.md [hunk 1]
```diff
@@ -20,6 +20,7 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 - The OpenEnclave CMake configuration now explicitly sets CMAKE_SKIP_RPATH to TRUE. This change should not affect fully static-linked enclaves.
 - oe_verify_attestation_certificate_with_evidence() has been deprecated because it has been deemed insufficient for security. Use the new, experimental oe_verify_attestation_certificate_with_evidence_v2() instead to generate a self-signed certificate for use in the TLS handshaking process.
 - In/out parameters in EDL now have the default count equals to one if the `count` attribute is not used.
+- The tool, `oecert`, has been renamed to `oegenerate`, as it generates not only certificates, but also reports and evidence in different formats.
```
Comment (mingweishih) [inline CHANGELOG.md:23]: Don't we publish the tool starting from this release? I think the log should be under the Added section and be rephrased.  For example,

Publish a debugging tool `oegenerate`
- The tool, currently under the tools/ directory, was originally named `oecert` under the tests/tools directory.
- The tool can be used to generate certificates, reports, and evidence in various formats.
- The tool is for debugging purposes and is not suitable for production use.

Comment (ryanhsu19) [reply CHANGELOG.md]: Sounds good, thanks!

**Pull request merged**